### ISSUE TITLE
Lint: three staticcheck QF1003 findings in nested_conditional_pattern.go

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#380]
## Summary

`golangci-lint` reports three staticcheck `QF1003` findings on `main` (`f99faa6`). They are all in `internal/parse/nested_conditional_pattern.go`:

| Location | Finding |
| --- | --- |
| `internal/parse/nested_conditional_pattern.go:329:4` | `QF1003: could use tagged switch on b` |
| `internal/parse/nested_conditional_pattern.go:340:4` | `QF1003: could use tagged switch on b` |
| `internal/parse/nested_conditional_pattern.go:363:4` | `QF1003: could use tagged switch on b` |

Each is an `if b == '<quote>' { ... } else if b == ')' { ... }` chain inside the single-quoted, ANSI-C-quoted and double-quoted arms of the quote-state switch. The code came from #225 and #258.

## Why CI does not report it

The `Trunk Check` job judges only lines a pull request changes (hold-the-line). Nobody has changed these lines since the findings appeared, so every pull request passes.
A full run shows them:

```
trunk check --all --no-fix --filter=golangci-lint2
✖ 3 lint issues
```

`golangci-lint run ./...` gives the same result: `golangci-lint` 2.12.2 under Go 1.26.7, the toolchain `go.mod` pins, with no repository `.golangci.yml`.
Those are the only findings in the repository.

Found during the #368 work (PR #378), which does not touch this file.

## Fix

Rewrite the three chains as tagged switches on `b`, for example:

```go
switch b {
case '\'':
	frame.quote = activeSourceUnquoted
case ')':
	noteQuotedPatternClose(frame, i)
}
```

This does not change behaviour, because the conditions are mutually exclusive byte comparisons.

## Acceptance

- `trunk check --all --no-fix --filter=golangci-lint2` reports no issues.
- `go test ./...` passes unchanged.
